modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/ConfigurationBasedElementModelFactory.java

```diff
@@ -592,24 +597,13 @@ private void handleInfrastructure(final ParameterModel paramModel, final DslElem
         return;
 
       case STREAMING_STRATEGY_PARAMETER_NAME:
-        ComponentIdentifier inMemoryStream = newIdentifier(NON_REPEATABLE_BYTE_STREAM_ALIAS, paramDsl.getPrefix());
-        ComponentIdentifier repeatableMemoryStream =
-            newIdentifier(REPEATABLE_IN_MEMORY_BYTES_STREAM_ALIAS, paramDsl.getPrefix());
-        ComponentIdentifier repeatableFileStream =
-            newIdentifier(REPEATABLE_FILE_STORE_BYTES_STREAM_ALIAS, paramDsl.getPrefix());
-
-        ComponentConfiguration streaming = null;
-        if (nested.containsKey(inMemoryStream)) {
-          streaming = nested.get(inMemoryStream);
-        } else if (nested.containsKey(repeatableMemoryStream)) {
-          streaming = nested.get(repeatableMemoryStream);
-        } else if (nested.containsKey(repeatableFileStream)) {
-          streaming = nested.get(repeatableFileStream);
-        }
+        Set<ComponentIdentifier> streaming =
+            Sets.newHashSet(newIdentifier(NON_REPEATABLE_BYTE_STREAM_ALIAS, CORE_PREFIX),
+                            newIdentifier(REPEATABLE_IN_MEMORY_BYTES_STREAM_ALIAS, CORE_PREFIX),
+                            newIdentifier(REPEATABLE_FILE_STORE_BYTES_STREAM_ALIAS, EE_PREFIX));
```

Comment:
what about: 
 `REPEATABLE_IN_MEMORY_OBJECTS_STREAM_ALIAS = "repeatable-in-memory-iterable";`
`REPEATABLE_FILE_STORE_OBJECTS_STREAM_ALIAS = "repeatable-file-store-iterable";`
`NON_REPEATABLE_OBJECTS_STREAM_ALIAS = "non-repeatable-iterable";`